client/ayon_core/hooks/pre_filter_farm_environments.py

```diff
@@ -0,0 +1,78 @@
+import copy
+import re
+
+from ayon_applications import PreLaunchHook, LaunchTypes
+from ayon_core.lib import filter_profiles
+
+
+class FilterFarmEnvironments(PreLaunchHook):
+    """Filter or modify calculated environment variables for farm rendering.
+
+    This hook must run last, only after all other hooks are finished to get
+    correct environment for launch context.
+
+    Implemented modifications to self.launch_context.env:
+    - skipping (list) of environment variable keys
+    - removing value in environment variable:
+        - supports regular expression in pattern
+        - doesn't remove env var if value empty!
+    """
+    order = 1000
+
+    launch_types = {LaunchTypes.farm_publish}
+
+    def execute(self):
+        data = self.launch_context.data
+        project_settings = data["project_settings"]
+        filter_env_profiles = (
+            project_settings["core"]["filter_farm_environment"])
+
+        if not filter_env_profiles:
+            self.log.debug("No profiles found for env var filtering")
+            return
+
+        task_entity = data["task_entity"]
+
+        filter_data = {
+            "hosts": self.host_name,
+            "task_types": task_entity["taskType"],
+            "tasks": task_entity["name"],
+            "folders": data["folder_path"]
+        }
+        matching_profile = filter_profiles(
+            filter_env_profiles, filter_data, logger=self.log
+        )
+        if not matching_profile:
+            self.log.debug("No matching profile found for env var filtering "
+                           f"for {filter_data}")
+            return
+
+        calculated_env = copy.deepcopy(self.launch_context.env)
+
+        calculated_env = self._skip_environment_variables(
+            calculated_env, matching_profile)
+
+        calculated_env = self._modify_environment_variables(
+            calculated_env, matching_profile)
+
+        self.launch_context.env = calculated_env
+
+    def _modify_environment_variables(self, calculated_env, matching_profile):
+        """Modify environment variable values."""
+        for env_item in matching_profile["replace_in_environment"]:
+            value = calculated_env.get(env_item["environment_key"])
+            if not value:
+                continue
+
+            value = re.sub(value, env_item["pattern"], env_item["replacement"])
+            calculated_env[env_item["environment_key"]] = value
```

Comment:
```suggestion
            key = env_item["environment_key"]
            value = calculated_env.get(key)
            if not value:
                continue

            value = re.sub(value, env_item["pattern"], env_item["replacement"])
            if value:
                calculated_env[key] = value
            else:
                calculated_env.pop(key)
            
```

Reply:
I was thinking about it and I am not sure this is right. But I am not sure, that current implementation (eg. there could be FOO with empty value) is.

Reply:
Empty value means that the env should not be set.

Reply:
What about use case where you are using env var placeholders in file paths? 
(path_to_licence: `//home/blabla/license{LICENSE_VARIANT}` )

Reply:
Not sure how is that connected. If output of
```python
value = re.sub(value, env_item["pattern"], env_item["replacement"])
```
is empty string, it means the environment variable is empty.

If you want to replace value in `"//home/blabla/license{LICENSE_VARIANT}"` you'll probably replace only `"{LICENSE_VARIANT}"`, in worst case scenario the value will be ``"//home/blabla/license"`` which is not `""`.

Reply:
Ok, updated.